src/Data/Entity.php

```diff
@@ -3,52 +3,60 @@
 
 abstract class Entity implements \JsonSerializable
 {
-    protected $properties = [];
-
     protected $id;
+    protected $createdAt;
+    protected $updatedAt;
+    protected $deleted = false;
 
-    private $data;
-
-    private $callMethods = ['get', 'set'];
+    public function getId()
+    {
+        return $this->id;
+    }
 
-    public function __construct($data = [])
+    public function setId($id)
     {
-        $this->data = $data;
+        $this->id = (int)$id;
     }
 
-    public function getId()
+    public function getCreatedAt()
     {
-        return $this->id;
+        return $this->createdAt;
     }
 
-    public function mapData(array $row)
+    public function setCreatedAt($createdAt)
     {
-        $this->data = $row;
+        $this->createdAt = strtotime($createdAt);
```

Comment:
The reason I like doing these setters/getters is that now it's super easy to modify a value before it gets passed into the entity.